src/ServiceControl.RavenDB/EmbeddedDatabase.cs

```diff
@@ -162,23 +175,75 @@ public async Task<IDocumentStore> Connect(CancellationToken cancellationToken)
 
         public async Task DeleteDatabase(string dbName)
         {
-            using var store = await EmbeddedServer.Instance.GetDocumentStoreAsync(new DatabaseOptions(dbName) { SkipCreatingDatabase = true });
+            using var store = await EmbeddedServer.Instance.GetDocumentStoreAsync(new DatabaseOptions(dbName)
+            {
+                SkipCreatingDatabase = true
+            });
             await store.Maintenance.Server.SendAsync(new DeleteDatabasesOperation(dbName, true));
         }
 
+        public async Task Stop(CancellationToken cancellationToken)
+        {
+            Logger.Debug("Stopping RavenDB server");
+            EmbeddedServer.Instance.ServerProcessExited -= OnServerProcessExited;
+
+            await shutdownTokenSource.CancelAsync();
+
+            // TODO: await EmbeddedServer.Instance.StopAsync(cancellationToken);
+
+            var processId = await EmbeddedServer.Instance.GetServerProcessIdAsync(cancellationToken);
+
+            var waitForCancellationTask = Task.Delay(Timeout.Infinite, cancellationToken);
+            var firstTask = await Task.WhenAny(
+                Task.Run(() => EmbeddedServer.Instance.Dispose(), cancellationToken),
+                waitForCancellationTask
```

Comment:
@ramonsmits I am trying to understand the reason for this code.
I thought we wanted to give a chance for RavenDB server to end on its own, and then if it takes a long time, then we resort to killing it, or am I misunderstanding?

Reply:
> I thought we wanted to give a chance for RavenDB server to end on its own, and then if it takes a long time, then we resort to killing it, or am I misunderstanding?

@johnsimons that is exactly what this code should do:

- Launch the dispose as a task, and the timer
- Wait for either to complete
- If its the timer task it ran out of time and we kill the child process

Reply:
@johnsimons it can take many minutes for the Raven process to finish but the dispose of their wrapper uses 30 seconds, which at the moment is longer that windows shutdown duration thus our delay is cancelled usually before the shutdown. 

If we kill it, and its already done... that is fine.